src/main/java/com/minecolonies/colony/Colony.java

```diff
@@ -515,110 +452,169 @@ public void updateSubscribers()
                     }
 
                     double distance = player.getDistanceSq(center);
-                    if (distance < MathUtils.square(Configurations.workingRangeTownHall + 16))
+                    if (distance < MathUtils.square(Configurations.workingRangeTownHall + 16D) ||
+                            (oldSubscribers.contains(player) && distance < MathUtils.square(Configurations.workingRangeTownHall * 2D)))
                     {
-                        //  Players become subscribers if they come within 16 blocks of the edge of the colony
-                        subscribers.add(player);
-                    }
-                    else if (oldSubscribers.contains(player) &&
-                             distance < MathUtils.square(Configurations.workingRangeTownHall * 2))
-                    {
-                        //  Players remain subscribers while they remain within double the colony's radius
+                        // Players become subscribers if they come within 16 blocks of the edge of the colony
+                        // Players remain subscribers while they remain within double the colony's radius
                         subscribers.add(player);
                     }
                 }
             }
         }
 
-        //  Determine if any new subscribers were added this pass
-        boolean hasNewSubscribers = false;
-        for (EntityPlayerMP player : subscribers)
-        {
-            if (!oldSubscribers.contains(player))
-            {
-                hasNewSubscribers = true;
-                break;
-            }
-        }
-
         if (!subscribers.isEmpty())
         {
+            //  Determine if any new subscribers were added this pass
+            boolean hasNewSubscribers = hasNewSubscribers(oldSubscribers, subscribers);
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Local variable 'hasNewSubscribers' could be declared final [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=pmd%3ALocalVariableCouldBeFinal)